okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java

```diff
@@ -191,12 +262,27 @@ private HttpUrl(String scheme, String username, String password, String host, in
     this.url = url;
   }
 
+  /** Returns this URL as a {@link URL java.net.URL}. */
   public URL url() {
-    throw new UnsupportedOperationException(); // TODO(jwilson).
+    try {
+      return new URL(url);
+    } catch (MalformedURLException e) {
+      throw new RuntimeException(e); // Unexpected!
+    }
   }
 
-  public URI uri() throws IOException {
-    throw new UnsupportedOperationException(); // TODO(jwilson).
+  /**
+   * Attempt to convert this URL to a {@link URI java.net.URI}. This method throws an unchecked
+   * {@link IllegalStateException} if the URL it holds isn't valid by URI's overly-stringent
```

Comment:
I want to explicitly call this out in the main part of the doc because it's surprising. Bad idea?